build.gradle

```diff
@@ -237,3 +247,98 @@ def getVersionMetadata() {
 		return "rev.${id}"
 	}
 }
+
+def ENV = System.getenv()
+
+def getBranch() {
+	def ENV = System.getenv()
+	if (ENV.GITHUB_REF) {
+		def branch = ENV.GITHUB_REF
+		return branch.substring(branch.lastIndexOf("/") + 1)
+	}
+
+	if (grgit != null) {
+		def head = grgit.head()
+		def id = head.abbreviatedId
+
+		// Flag the build if the build tree is not clean
+		if (!grgit.status().clean) {
+			id += "-dirty"
+		}
+	}
+
+	def branch = grgit.branch.current().name
+	return branch.substring(branch.lastIndexOf("/") + 1)
+}
+
+
+curseforge {
+	if (ENV.CURSEFORGE_TOKEN) {
+		apiKey = ENV.CURSEFORGE_TOKEN
+	}
+
+	project {
+		id = '470001'
+		changelog = "https://github.com/NoComment1105/Iris/tree/trunk/docs/changelogs/${project.mod_version}summary.md"
+		releaseType = 'release'
+		relations {
+			requiredDependency 'sodium'
+		}
+		addGameVersion "$project.minecraft_version"
+		addGameVersion 'Fabric'
+
+		mainArtifact(remapJar) {
+			displayName = "Iris $project.mod_version for Minecraft $project.minecraft_version"
+		}
+		afterEvaluate {
+			uploadTask.dependsOn(remapJar)
+		}
+	}
+	options {
+		forgeGradleIntegration = false
+	}
+}
+
+import com.modrinth.minotaur.TaskModrinthUpload
+
+task modrinth(type: TaskModrinthUpload, dependsOn: remapJar) {
+	onlyIf {
+		ENV.MODRINTH_TOKEN
+	}
+
+	token = ENV.MODRINTH_TOKEN
+	projectId = "5XACEUIs"
+	versionNumber = "mc$project.mod_version"
+	versionName = "Iris $project.mod_version for Minecraft $project.minecraft_version"
+	releaseType = 'release'
+	changelog = "https://github.com/NoComment1105/Iris/tree/trunk/docs/changelogs/${project.mod_version}summary.md"
+
+	uploadFile = remapJar
+
+	addGameVersion(project.minecraft_version)
+	addLoader('fabric')
+}
+
+
+import org.kohsuke.github.GHReleaseBuilder
+import org.kohsuke.github.GitHub
+
+task github(dependsOn: remapJar) {
+	onlyIf {
+		ENV.GITHUB_TOKEN
+	}
+
+	doLast {
+		def github = GitHub.connectUsingOAuth(ENV.GITHUB_TOKEN as String)
+		def repository = github.getRepository(ENV.GITHUB_REPOSITORY)
+
+		def releaseBuilder = new GHReleaseBuilder(repository, version as String)
+		releaseBuilder.name("Iris $project.mod_version for Minecraft $project.minecraft_version")
+		releaseBuilder.body("https://github.com/NoComment1105/Iris/tree/trunk/docs/changelogs/${project.mod_version}summary.md")
+		releaseBuilder.commitish(getBranch())
```

Comment:
Is this needed? Seems like we're converting from commit ID to branch, would it be more direct to just pass the commit ID?

Reply:
Possibly yes, this is left from when I was trying to get it to loop through all 3 branches to upload, however I'm going to change the approach for it.

Reply:
Turns out it's not needed at all as by default it gets the branch name